testplan/common/config/base.py

```diff
@@ -74,12 +83,20 @@ def get_key_str(option):
         """Will be used for getting name from ConfigOption keys."""
         return option._schema if isinstance(option, Optional) else option
 
-    target_raw_keys = {get_key_str(k) for k in target}
+    target_key_mapping = {get_key_str(k): k for k in target}
     source_key_mapping = {get_key_str(k): k for k in source}
 
-    for raw_key, key in source_key_mapping.items():
-        if raw_key not in target_raw_keys:
-            target[key] = source[key]
+    for raw_key, source_key in source_key_mapping.items():
+        if raw_key not in target_key_mapping:
+            target[source_key] = source[source_key]
+        else:
+            target_key = target_key_mapping[raw_key]
+            if isinstance(source_key.default, DefaultValueWrapper) and \
+                    source_key.default.low_precedence is not None and \
+                    isinstance(target_key.default, DefaultValueWrapper) and \
+                    target_key.default.low_precedence is None:
+                target_key.default.low_precedence = \
+                        source_key.default.low_precedence
```

Comment:
I think it's better write like this to be safe:
    value = target.pop(target_key)
    target_key.default.low_precedence = source_key.default.low_precedence 
    target[target_key] = value

Reply:
This piece of logic deserves some comment to explain its behavior: if inherited class re-define an option, and its low_precedence is default to None, this code will fetch the low_precedence flag from the option of the same name in parent class. (personally i think this is too much automation... but you have a point that when user set low_presedence=True he will want this to propagate down, let's see what John thinks)